csharp/ql/src/utils/modeleditor/ModelEditor.qll

```diff
@@ -129,3 +131,12 @@ private string qualifiedTypeName(string namespace, Type t) {
 private string qualifiedCallableName(string namespace, string type, Callable c) {
   exists(string name | hasQualifiedMethodName(c, namespace, type, name) | result = name)
 }
+
+/** A file that doesn't contain tests itself, but is only used in tests. */
+class TestSupportFile extends File {
```

Comment:
Unless you think it will become important to distinguish between `TestFile` and `TestSupportFile`, then I think you should just let `TestSupportFile` extend `TestFile` instead (in your case `TestFile` just means "test related files").

Reply:
I don't think it will be important to distinguish between the two. Would extending `TestFile` in here not apply to any other queries? I'm not entirely sure how abstract classes work with imports etc.

Reply:
It will only apply to queries that import `ModelEditor.qll`.
Typically one does a bi-directional import, if you want it to apply everywhere.

Reply:
Thanks, makes sense, then I'll change this to extend `TestFile`.

Reply:
I'm now getting a CodeQL alert for doing this (see [this alert](https://github.com/github/codeql/security/code-scanning/87290)). Can I just dismiss this alert?

Reply:
Uh, sorry yes - that is considered bad practice as there already exists some bi-directional imports. So let's not do it like that.
I will add another suggestion.